var/spack/repos/builtin/packages/nf-wave-cli/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import platform
+
+from spack.package import *
+
+
+class NfWaveCli(Package):
+    """Command line tool for Wave containers provisioning service."""
```

Comment:
Would it be difficult to build from sources (and have a single version directive for each release)?

Reply:
It is a Java application, it requires a JDK and Gradle to compile. 

However, to generate these portable native image executables the JDK requirements  are stricter (I am not an expert). We typically use GraalVM 21.x (https://github.com/oracle/graal).  Adding Spack support for GraalVM seems out of scope (and effort 😅 ) for this PR.   
An alternative approach I could take to have a (still pre-compiled) single recipe is to instead download the JAR file and create a shell script that launches it with the Java runtime.

Though I agree that a single build from source would be nicer,  note the application is not performance critical, it is a utility that can complement the use of Nextflow.

Keen on your thoughts!

Reply:
Let's revisit this in another PR then :wink: